### PR TITLE
Fuzz test the worker queue

### PR DESCRIPTION
**Pull request by gatesn**
And fix the edge-case of empty task factories

### CODE REVIEW

File: vortex-scan/src/work_queue.rs [hunk 1]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+    use std::thread;
+    use std::time::Duration;
+    use vortex_error::vortex_bail;
+
+    /// Fuzz test with varying numbers of workers and factories
+    #[test]
+    fn fuzz_worker_factory_combinations() {
+        for _ in 0..100 {
+            let mut rng = rand::rng();
+            let num_factories = rng.random_range(1..=20);
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: Should we change to exclusive ranges? [1, 20)

File: vortex-scan/src/work_queue.rs [hunk 2]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+            let num_workers = rng.random_range(1..=8);
+            let tasks_per_factory = rng.random_range(0..=100);
+
+            println!(
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: Do we want print statements in the test suite?

File: vortex-scan/src/work_queue.rs [hunk 3]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+                failure_rate,
+                max_delay_ms,
+            );
+        }
+    }
+
+    /// Fuzz test with workers joining and leaving at random times
+    #[test]
+    fn fuzz_dynamic_worker_lifecycle() {
+        for _ in 0..30 {
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: Do we want to unify the iteration counts across tests and define it as a constant?

File: vortex-scan/src/work_queue.rs [hunk 4]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+                ),
+                2 => test_mixed_factory_sizes(rng.random_range(10..=30), rng.random_range(3..=6)),
+                _ => test_empty_factories_mixed(rng.random_range(5..=20), rng.random_range(2..=4)),
+            }
+        }
+    }
+
+    /// Test edge cases with empty or single-task scenarios
+    #[test]
+    fn fuzz_edge_cases() {
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: Ehm wat? 😅

Comment (0ax1) [reply vortex-scan/src/work_queue.rs]: Are we randomizing which tests to invoke?

File: vortex-scan/src/work_queue.rs [hunk 5]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+
+    // Helper function to create a factory that produces a specific number of tasks
+    fn create_factory(
+        task_count: usize,
+        id: usize,
+        delay_ms: u64,
+        should_fail: bool,
+    ) -> TaskFactory<(usize, usize)> {
+        Box::new(move || {
+            if delay_ms > 0 {
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs:290]: I'd say let's drop the `delay_ms > 0`. Just pass 0 to sleep then.

File: vortex-scan/src/work_queue.rs [hunk 6]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+    }
+
+    fn test_concurrent_processing(
+        num_factories: usize,
+        num_workers: usize,
+        tasks_per_factory: usize,
+    ) {
+        let mut factories = Vec::new();
+        for i in 0..num_factories {
+            factories.push(create_factory(tasks_per_factory, i, 0, false));
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: `collect`

File: vortex-scan/src/work_queue.rs [hunk 7]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+                    barrier.wait();
+                    let mut iterator = queue.new_iterator();
+                    let mut local_count = 0;
+
+                    while let Some(task_result) = iterator.next() {
+                        match task_result {
+                            Ok(task) => {
+                                // Track this task to detect duplicates
+                                {
+                                    let mut tracking = task_tracking.lock().unwrap();
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs]: As we don't care about perf in the test, let's drop the extra scope for readability.

File: vortex-scan/src/work_queue.rs [hunk 8]
```diff
@@ -148,3 +156,521 @@ impl<T> Iterator for WorkQueueIterator<T> {
...
+            handle.join().unwrap();
+        }
+
+        let total_processed = processed_count.load(SeqCst);
+        assert_eq!(total_processed, expected_total);
+
+        println!("✓ Dynamic workers processed {} tasks", total_processed);
+    }
+
+    fn test_many_small_factories(num_factories: usize, num_workers: usize) {
```
Comment (0ax1) [inline vortex-scan/src/work_queue.rs:440]: `many` and `small` sounds like other functions should just be parametrized.
